### PR TITLE
Replace banner.txt with new logo

### PR DESCRIPTION
**Pull request by making**
#2737

image

### CODE REVIEW

File: zipkin-server/src/main/resources/banner.txt [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
-                                       **
-                                       **
-
-
-             *****      **     *****     ** **       **     **   **
-               **       **     **  *     ***         **     **** **
-              **        **     *****     ****        **     **  ***
-             ******     **     **        **  **      **     **   **
+${AnsiColor.RED}
+                           o
```
Comment (jeqo) [inline zipkin-server/src/main/resources/banner.txt]: nit: I feel is missing one dot on the left

```suggestion
                          oo
```

Comment (making) [reply zipkin-server/src/main/resources/banner.txt]: Fixed :)

**Comment by shakuzen** [comment]
We might want to wait on merging this for the outcome of the discussion on openzipkin vs zipkin in the logo.

**Comment by codefromthecrypt** [comment]
actually definitely the server should just be zipkin. I think it would be
certainly incorrect here.

**Comment by making** [comment]
Updated

image

**Review by codefromthecrypt — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
ahead of the curve huh? :D Screenshot 2019-08-08 at 1 21 31 PM

**Comment by making** [comment]
I tested `banner.txt` in a latest spring boot app.

**Review by tacigar — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
if you can verify this is correct still with the new logo content: https://github.com/openzipkin/openzipkin.github.io/pull/142 then later or now we still need to update lens images

**Comment by codefromthecrypt** [comment]
only difference is if we think red or orange, I guess

**Review by jeqo — APPROVED** [review]
sadly, seems that orange is not available https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/ansi/AnsiColor.html but I will let @making confirm if possible.
we might fallback to white as well.

**Comment by making** [comment]
@jeqo @adriancole as far as I know, orange is not supposed.

**Comment by codefromthecrypt** [comment]
I think black/white/grey is better than red when orange unsupported. Red is a more aggressive color.

can you raise an issue to support orange :)

**Comment by making** [comment]
I'll try to implement a new `org.springframework.boot.ansi.AnsiElement` so that it supports 256 colors.

Which color do you prefer? :)
image

You can get this table by

```
for i in $(seq 1 255);do printf "%03d \033[38;5;${i}mZipkin\033[0;00m\t" $i; if [ $(( $i % 10 )) -eq 0 ];then echo;fi ;done;echo
```

**Comment by making** [comment]
I think I managed to change the color :)

image

Let me update the source code later

**Comment by jorgheymans** [comment]
> I think I managed to change the color :)
> 
> image
> 
> Let me update the source code later

Cool ! Will it work in a windows shell as well ?

**Comment by making** [comment]
I've just pushed the updated banner in 4ca0ae3e5dcc14b420e5424bccf8b82fa91cd6b6.

image


@jorgheymans I don't have a Windows environment. Could you please give it a try?

**Comment by making** [comment]
In 4ca0ae3e5dcc14b420e5424bccf8b82fa91cd6b6
I added `ZipkinAnsi256Color`, `ZipkinAnsi256PropertySource` and `ZipkinBanner`, then configured the banner programmatically as described in the boot [doc](https://docs.spring.io/spring-boot/docs/2.1.8.RELEASE/reference/html/boot-features-spring-application.html#boot-features-banner) in order to support 256 colors that is not available in Spring Boot out of the box now.
I'm thinking to contribute to Spring Boot but it wouldn't come to Spring Boot 2.1.x. So Zipkin needs to have these classes until a feature version.

File: zipkin-server/src/main/java/zipkin2/server/internal/banner/ZipkinAnsi256Color.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.server.internal.banner;
+
+import org.springframework.boot.ansi.AnsiElement;
+
+public class ZipkinAnsi256Color implements AnsiElement {
```
Comment (codefromthecrypt) [inline zipkin-server/src/main/java/zipkin2/server/internal/banner/ZipkinAnsi256Color.java:23]: add a TODO to delete on spring boot 2.2 if that's when this is scheduled to be normally possible.

Comment (making) [reply zipkin-server/src/main/java/zipkin2/server/internal/banner/ZipkinAnsi256Color.java:23]: Updated in edebc31428c9a3551181aa053ecb008b9dd2666e

**Comment by codefromthecrypt** [comment]
Will merge this on green.

Added issue to update lens https://github.com/openzipkin/zipkin/issues/2806

**Review by codefromthecrypt — APPROVED** [review]
thanks again!

**Comment by making** [comment]
It's been merged in Spring Boot 2.2 :)
https://github.com/spring-projects/spring-boot/pull/18264

**Comment by codefromthecrypt** [comment]
nice to change your shoes on occasion!

**Pull request merged**